nixos/modules/services/mail/public-inbox.nix

```diff
@@ -0,0 +1,557 @@
+    };
+    settings = mkOption {
+      description = "Settings for the public-inbox config file.";
+      default = {};
+      type = types.submodule {
+        freeformType = gitIni.type;
+        options.publicinbox = mkOption {
+          default = {};
+          description = "public-inbox configuration.";
+          type = types.submodule {
+            freeformType = iniAttrs;
+            options.css = mkOption {
+              type = with types; listOf str;
+              default = [];
+              description = "The local path name of a CSS file for the PSGI web interface.";
+            };
+            options.nntpserver = mkOption {
+              type = with types; listOf str;
+              default = [];
+              example = [ "nntp://news.public-inbox.org" "nntps://news.public-inbox.org" ];
+              description = "NNTP URLs to this public-inbox instance";
+            };
+            options.wwwlisting = mkOption {
+              type = with types; enum [ "all" "404" "match=domain" ];
+              default = "404";
+              description = ''
+                Controls which lists (if any) are listed for when the root
+                public-inbox URL is accessed over HTTP.
+              '';
+            };
+          };
+        };
+        options.publicinboxmda = mkOption {
+          default = {};
+          description = "mailbox delivery agent";
+          type = types.submodule {
+            freeformType = iniAttrs;
+            options.spamcheck = mkOption {
+              type = with types; enum [ "spamc" "none" ];
+              default = "none";
+              description = ''
+                If set to spamc, ${manref "public-inbox-watch" 1} will filter spam
+                using SpamAssassin.
+              '';
+            };
+          };
+        };
+        options.publicinboxwatch = mkOption {
+          default = {};
+          description = "mailbox watcher";
+          type = types.submodule {
+            freeformType = iniAttrs;
+            options.spamcheck = mkOption {
+              type = with types; enum [ "spamc" "none" ];
+              default = "none";
+              description = ''
+                If set to spamc, ${manref "public-inbox-watch" 1} will filter spam
+                using SpamAssassin.
+              '';
+            };
+            options.watchspam = mkOption {
+              type = with types; nullOr str;
+              default = null;
+              example = "maildir:/path/to/spam";
+              description = ''
+                If set, mail in this maildir will be trained as spam and
+                deleted from all watched inboxes
+              '';
+            };
+          };
+        };
+        options.coderepo = mkOption {
+          default = {};
+          description = "code repositories";
+          type = types.attrsOf (types.submodule {
+            freeformType = types.either (types.attrsOf iniAtom) iniAtom;
```

Comment:
Although this won't work here. However, you can get access to the ini atom type with `gitIni.type.functor.wrapped.functor.wrapped`

I'm a bit confused here though, how can this be `attrsOf iniAtom`? Wouldn't that be three nestings then? There is the `coderepo` attribute, then we have an `attrsOf`, and then another `attrsOf`. And INI only supports 2-level nestings.

Reply:
Here's an example:
```
services.public-inbox = {
  settings.coderepo = { # first types.attrsOf
    sourcephile-txt = { # second types.attrsOf
      dir = "/var/lib/gitolite/repositories/sourcephile-txt.git";
      cgitUrl = "https://code.sourcephile.fr/sourcephile-txt.git";
    };
  };
};
```
Generating:
```
[coderepo "sourcephile-nix"]
        cgitUrl = https://code.sourcephile.fr/sourcephile-nix.git
        dir = /var/lib/gitolite/repositories/sourcephile-nix.git
```
Since they're is no `[coderepo]` section, I've removed the useless `iniAtom` in the second branch of `types.either`.